src/Planners/src/TimeVaryingDCMPlanner.cpp

```diff
@@ -230,43 +211,31 @@ struct TimeVaryingDCMPlanner::Impl
         this->opti.subject_to(omega(Sl(), 0) == this->optiParameters.omegaInitialValue);
         this->opti.subject_to(dcm(Sl(), 0) == this->optiParameters.dcmInitialPosition);
 
-        // initial value of the DCM dynamics
-        dcmDynamicalSystem->setState({dcm(Sl(), 0), omega(Sl(), 0)});
-        dcmDynamicalSystem->setControlInput({vrp(Sl(), 0), omegaDot(Sl(), 0)});
 
-        TimeVaryingDCMPlannerDynamics::StateDerivativeType stateDerivative;
-        dcmDynamicalSystem->dynamics(0, stateDerivative);
+        auto initialDcmVelocity = (omega(0) - omegaDot(0) / omega(0)) * (dcm(Sl(), 0) - vrp(Sl(), 0));
+        auto initialOmegaVelocity = omegaDot(0);
 
-        const auto& [initialDcmVelocity, initialOmegaVelocity] = stateDerivative;
         this->opti.subject_to(casadi::MX::vertcat({initialDcmVelocity, initialOmegaVelocity})
                               == casadi::MX::vertcat({this->optiParameters.dcmInitialVelocity, 0}));
 
         // final values
         this->opti.subject_to(omega(Sl(), -1) == this->optiParameters.omegaFinalValue);
         this->opti.subject_to(dcm(Sl(), -1) == this->optiParameters.dcmFinalPosition);
 
+        auto finalDcmVelocity = (omega(-1) - omegaDot(-1) / omega(-1)) * (dcm(Sl(), -1) - vrp(Sl(), -1));
```

Comment:
Just to understand, what is the difference between ``Sl()`` and ``-1``?

Reply:
`Sl()` takes the entire slice `-1` the last element. For instance 
`vrp(Sl(), -1))` can be seen as `vrp(:, end)`

Reply:
Ah I see, thanks